app/models/round.rb

```diff
@@ -121,6 +121,42 @@ def name
     Round.name_from_attributes(event, round_type)
   end
 
+  def registrations
+    if number == 1
+      Registration.joins(:registration_competition_events)
+                  .where(
+                    competition_id: competition_event.competition_id,
+                    competing_status: 'accepted',
+                    registration_competition_events: { competition_event_id: competition_event_id }
+                  ).includes([:user])
+    else
+      previous_round = Round.joins(:competition_event).find_by(competition_event: { competition_id: competition_event.competition_id, event_id: event.id }, number: number - 1)
```

Comment:
```suggestion
      previous_round = Round.joins(:competition_event).find_by(competition_event: { competition_id: competition_event.competition_id, event_id: competition_event.event_id }, number: number - 1)
```

Reply:
you can even do `Round.joins(:competition_event).find_by(competition_event: competition_event)`